realm/src/androidTest/java/io/realm/RealmJsonTest.java

```diff
@@ -121,6 +123,23 @@ public void testCreateObjectFromJson_dateAsString() throws JSONException {
         assertEquals(new Date(1000), obj.getColumnDate());
     }
 
+    public void testCreateObjectFromJson_dateAsStringTimeZone() throws JSONException {
+        JSONObject json = new JSONObject();
+
+        // Oct 03 2015 14:15
+        json.put("columnDate", "/Date(1443854733376+0800)/");
+
+        testRealm.beginTransaction();
+        testRealm.createObjectFromJson(AllTypes.class, json);
+        testRealm.commitTransaction();
+
+        AllTypes obj = testRealm.allObjects(AllTypes.class).first();
+        Calendar cal = GregorianCalendar.getInstance();
```

Comment:
To ensure this tests work on all devices you need to explicitly set the timezone.

```
Calendar.getInstance(TimeZone.getTimeZone("Some +8 timezone"));
```